### PR TITLE
docs: replace unsupported value 'string' in controls

### PR DESCRIPTION
**Pull request by joaonunomota**
## What I did

- Replaced all occurrences of `string` as the control type in docs with `text`, since `string` is not an accepted value.
- Replaced the use of `control: 'string'` in a test to avoid confusion, the test passes after updating the input and expected output.

## Checklist

- [x] Make sure your changes are tested (stories and/or unit, integration, or end-to-end tests)
- [x] Make sure to add/update documentation regarding your changes
- [x] If you are deprecating/removing a feature, make sure to update
      [MIGRATION.MD](https://github.com/storybookjs/storybook/blob/next/MIGRATION.md)

#### Maintainers

- [ ] When this PR is ready for testing, make sure to add `ci:normal`, `ci:merged` or `ci:daily` GH label to it to run a specific set of sandboxes. The particular set of sandboxes can be found in `code/lib/cli/src/sandbox-templates.ts`
- [ ] Make sure this PR contains **one** of the labels below.

`["cleanup", "BREAKING CHANGE", "feature request", "bug", "build", "documentation", "maintenance", "dependencies", "other"]`

### CODE REVIEW

**Review by tmeasday — APPROVED** [review]
Thanks!

**Review by jonniebigodes — APPROVED** [review]
@joaonunomota appreciate the time you took to put together this pull request and update the documentation and tests 🙏 ! From my end, documentation-wise, it's all good my end, I'll defer to the remainder maintainers for their input, and hopefully, we'll get this merged soon.

Hope you have a great week

Stay safe

**Review by ndelangen — APPROVED** [review]

**Pull request merged**
